### PR TITLE
Adding support for cpp-logger and dlio profiler.

### PR DESCRIPTION
**Pull request by hariharan-devarajan**
We can make our loggers and profiler configurable.

- [x] Build optional building for FLUX and CPP Logger building
- [x] Build optional building for DLIO Profiler. This would be used later by Caliper PR.
- [x] Add Profiler annotations within the code.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -23,7 +23,6 @@ include_directories(${CMAKE_BINARY_DIR}/include)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src) # required for private includes
 # Do we want to use RPath to simplify linking
 set(CMAKE_MACOSX_RPATH 1)
```
Comment (hariharan-devarajan) [inline CMakeLists.txt:25]: test comment

**Review by hariharan-devarajan — COMMENTED** [review]
Test review

**Comment by hariharan-devarajan** [comment]
@JaeseungYeom The CI are hanging because we our argparse is not working.

**Review by JaeseungYeom — APPROVED** [review]
This looks good to me. We will integrate caliper after this merge.

**Pull request merged**